### PR TITLE
Moved PollSettings from Google.LongrunningOperations

### PR DESCRIPTION
**Pull request by jskeet**
Also moved the code to perform the repeated polling, so
that it's easy to use from other code.

PollSettings is now mutable, like all other settings classes - but
as normal, it should not be mutated when in use.

### CODE REVIEW

**Comment by jskeet** [comment]
(Unit tests TBD - I basically need to extract them from the Operations code, but there they're attached to server code...)

File: src/Google.Api.Gax/PollSettings.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    public class PollSettings
+    {
+        /// <summary>
+        /// Settings to override the <see cref="OperationsClient"/> call settings
+        /// when polling, if any.
+        /// </summary>
+        public CallSettings CallSettings { get; set; }
+
+        /// <summary>
+        /// How long to wait before giving up. This may be null, which is equivalent to <see cref="Expiration.None"/>
```
Comment (chrisdunelm) [inline src/Google.Api.Gax/PollSettings.cs]: I do wonder if Expiration should be a struct

**Comment by jskeet** [comment]
Now with tests :) Will merge when green.

**Comment by chrisdunelm** [comment]
Great :)

**Pull request merged**